src/merkle_set.rs

```diff
@@ -91,6 +91,8 @@ fn radix_sort(range: &mut [[u8; 32]], depth: u8) -> ([u8; 32], NodeType) {
             // if every bit is identical, we have a duplicate value
```

Comment:
Question about this comment: Why does depth 255 having only a single child imply that we have an all-right or all-left tree ("if every bit is identical") ?

Are we expected to have returned already in the other cases?

Reply:
we build the tree by comparing the hash, bit-by-bit, from the left to right. As long as a node has more than 1 node, we keep "splitting" them based on the next bit. If you end up at the *last* bit (i.e. bit 255) and they are the same, we know the hashes are duplicates.

I believe it really should be an error when this happens, because it suggests that we're trying to build a block with duplicate spends (which is forbidden by other rules). However, the original implementation in python allowed (and ignored) duplicates, so to be compatible with existing tests, I also ignore them here

Reply:
Ah, right. Hash equality is like an escape condition for this type of search. Thanks.

> I believe it really should be an error when this happens, because it suggests that we're trying to build a block with duplicate spends (which is forbidden by other rules). However, the original implementation in python allowed (and ignored) duplicates, so to be compatible with existing tests, I also ignore them here

You should put this in a comment, since this merkle set implementation is embedded in this specific project, (and not a general library). We might want to update the code when the Python implementation is no longer used.